design/one-pager-release-process.md

```diff
@@ -0,0 +1,172 @@
+there are also some infrastructure stacks that the Crossplane community currently owns and is responsible for releasing, such as [`stack-gcp`](https://github.com/crossplaneio/stack-gcp).
+Since these stack repositories currently also take advantage of the build submodule, they will also follow a very similar process as Crossplane and receive a similar set of improvements.
+
+Recall that a Stack is simply just an [OCI image](https://github.com/crossplaneio/crossplane/blob/master/design/design-doc-stacks.md#stack-package-format).
+Therefore, the build and publishing process can follow essentially the same process that Crossplane uses.
+Other functionality in the build submodule, such as creating a Helm chart, is not necessary for Stacks, but relevant functionality should definitely be reused.
+For example, building and publishing the stack package, tagging and creating a release branch, promoting the release, etc.
+
+The GCP, AWS, and Azure stacks will all receive new `tag` and `promote` pipelines similar to Crossplane, as well as receive a fix for [#330](https://github.com/crossplaneio/crossplane/issues/330).
+
+Note that each Stack can be released independently and on their own schedule, but for now they will follow the same monthly release schedule as Crossplane so the community can receive frequent updates.
+
+## Issues and Unknowns
+
+### Upgrade
+
+While Crossplane and Stacks do not have an officially supported upgrade path, this proposal for more frequent updates may cause some pain within the community.
+This isn't ideal but can be acceptable while we are still in alpha.
+We should prioritize working out our upgrade story to remove this obstacle and allow the community to take a hard dependency on running Crossplane across versions.
+
+### Commits after tagging
+
+Currently, our tagging logic performs the tag on the HEAD commit of `master` and then creates the release branch.
+This means that the very first build out of the release branch will be the official minor release, since you cannot re-tag in `git` without essentially rewriting history.
+If we find an issue within final testing of the release build, we'd essentially need to either rewrite the tag history or run a patch build to release instead.
+
+This issue can also happen when we update the [example yaml files](https://github.com/crossplaneio/crossplane/blob/release-0.3/cluster/examples/stacks/stack-aws.yaml#L12) to embed the current release version in them.
+We would not want to make this update in `master` before tagging and branching, since that change is not relevant in `master`,
+but we also don't want to do it in the release branch after we've already tagged.
+One proposal would be to separate the tagging and branching functionality so that we branch from `master` first,
+make any updates/commits in the new release branch, and then finally tag the final commit within the release branch (not `master`).
```

Comment:
I like this strategy. Have we thought at all about potentially cutting the release branch some period of time (it seems like a week might be appropriate given our cadence) before the release date and instituting a "code freeze" (i.e. "feature complete") phase?

Reply:
Sounds good guys, I'll update the process so that we are tagging after we have branched and vetted the release candidate.  Yes @hasheddan, a code freeze period is something I want to incorporate into the concrete schedule I'll be adding soon.  A code freeze period would actually work well in the following flow:

1. `master` has all the features we're expecting to include in the release
1. create the release branch, but do not tag the release yet
1. "code freeze" period starts but is only applicable to the release branch.  `master` can continue taking changes.
1. only fixes to stabilize the release are accepted into the release branch during code freeze
1. testing and vetting occurs, every thing looks good.
1. tag the final commit in the release branch, e.g. `v0.4.0.`.
1. publish the release
1. profit 💰